### PR TITLE
Add CSS configurable option and change the Product Summary appearance

### PR DESCRIPTION
**Pull request by estacioneto**
#### What is the purpose of this pull request?

1. Change the Product Summary style to match with the designer specifications and the VTEX Tachyons guideline
2. Add some default (placeholder) data so the user can see the summary working without the real data.
3. Add CSS classes in the component so the components can be configurable.

#### What problem is this solving?

1. Style definition according to the designer definition. 
2. Without a product, the card was showing only the "loading" message.
3. There was no way to configure the component's design from the outside.

#### How should this be manually tested?
[Click here to access the workspace](https://estacio--storecomponents.myvtex.com/)
#### Screenshots or example usage
image

#### Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Requires change to documentation, which has been updated accordingly.

### CODE REVIEW

File: react/package.json [hunk 1]
```diff
@@ -6,7 +6,8 @@
     "test": "jest"
   },
   "dependencies": {
-    "@vtex/styleguide": "2.0.0-rc.35"
+    "@vtex/styleguide": "2.0.0-rc.35",
```
Comment (brunojdo) [inline react/package.json]: `rc.39`

File: react/ProductSummary.js [hunk 2]
```diff
@@ -1,11 +1,14 @@
 import React, { Component } from 'react'
 import PropTypes from 'prop-types'
-import Button from '@vtex/styleguide/lib/Button'
 import { FormattedMessage } from 'react-intl'
 
+import Card from '@vtex/styleguide/lib/Card'
```
Comment (brunojdo) [inline react/ProductSummary.js]: We decided don't use the Card component.

File: react/Price.js [hunk 3]
```diff
@@ -32,33 +32,32 @@ class Price extends Component {
...
-          <div className="pv1">
+          <div className="pv1 f6 normal">
             {showLabels &&
-              <div className="dib">
+              <div className="vtex-product-summary__price-list-price-label dib">
                 <FormattedMessage id="pricing.from" />
               </div>
             }
-            <div className="dib strike ph2">
+            <div className="vtex-product-summary__price-list-price dib strike ph2">
```
Comment (brunojdo) [inline react/Price.js]: `vtex-price`
`vtex-price-list__label`

Seems a little weird, don't you think? `vtex-price-list__label-price`

File: react/DiscountBadge.js [hunk 4]
```diff
@@ -14,14 +14,15 @@ class DiscountBadge extends Component {
...
     const { listPrice, sellingPrice, label } = this.props
     const percent = this.calculateDiscountTax(listPrice, sellingPrice)
-    return (percent !== 0) &&
-    <div className="relative dib">
-      <div className="f7 dark-gray absolute right-0 pa2-s bg-white">
-        { label === '' && '-' }
-        <FormattedNumber value={percent} style="percent" /> { label }
+    return percent ? (
+      <div className="relative dib">
+        <div className="f7 dark-gray absolute right-0 pa2-s bg-white">
```
Comment (brunojdo) [inline react/DiscountBadge.js]: Add `vtex-discount-badge`

File: react/ProductSummary.js (hunk 2 above)
Comment (estacioneto) [reply react/ProductSummary.js]: We decided to show something when hover. That was done

File: react/ProductName.js [hunk 5]
```diff
@@ -9,7 +9,7 @@ class ProductName extends Component {
     const { name, referenceCode, skuName, brandName } = this.props
 
     return (
-      <div className="f5 gray db tc overflow-hidden no-underline">
+      <div className="vtex-product-summary__name overflow-hidden">
```
Comment (brunojdo) [inline react/ProductName.js]: @vitoriaHeliane  will use this component. `vtex-product-name`

File: react/ProductFactory.js [hunk 6]
```diff
@@ -0,0 +1,19 @@
+import _ from 'lodash'
```
Comment (klzns) [inline react/ProductFactory.js]: O uso recomendado do `lodash` é sempre importar o modulo especifico. Importar dessa forma pode fazer com que a aplicação fique muito pesada, pq vai carregar o lodash inteiro.

```diff
- import _ from 'lodash'
+ import random from 'lodash/random'
```

Isso é muito importante.

File: react/ProductSummary.js [hunk 7]
```diff
@@ -40,60 +44,68 @@ class ProductSummary extends Component {
...
-                }
+                      installments={product.installments}
+                      installmentPrice={product.installmentPrice}
+                      showListPrice={showListPrice}
+                      showLabels={showLabels}
+                      showInstallments={showInstallments} />
+                  </div>
+                </div>
+                <div className="pv2">
+                  <div className={`${(!showButtonOnHover || (showButtonOnHover && this.state.isHovering)) ? 'db' : 'dn'}`}>
```
Comment (leticiawanderley) [inline react/ProductSummary.js]: Essa expressão pode ser simplificada para `!showButtonOnHover || this.state.isHovering`

Comment (estacioneto) [reply react/ProductSummary.js]: Tinha sido simplificada no PR do botão de detalhes, mas foi fechado :cry:

**Pull request merged**